### PR TITLE
core: several cleanups

### PR DESCRIPTION
**Pull request by yuwata**
~~Mostly no functional change. This only changes `unit_kill_context()` in a corner case.~~

No functional change, just refactoring.

### CODE REVIEW

**Comment by yuwata** [comment]
Dropped the commit that changes behavior. Now, this should not change any functionalities.

File: src/core/unit.c [hunk 1]
```diff
@@ -4000,6 +4000,48 @@ static int kill_or_sigqueue(PidRef* pidref, int signo, int code, int value) {
...
+
+        _cleanup_free_ char *comm = NULL;
+        (void) pidref_get_comm(pidref, &comm);
+
+        r = kill_or_sigqueue(pidref, signo, code, value);
+        if (r == -ESRCH)
+                return 0;
+        if (r < 0) {
+                /* Report this failure both to the logs and to the client */
+                if (!sd_bus_error_is_set(error))
```
Comment (poettering) [inline src/core/unit.c]: hmm, this is a bit weird. so far we never looked into this, we only wrote into this on error.

i'd avoid this, and instead just pass NULL for error as argument to unit_kill_one() unless ret == 0. I think that's more elegant and keeps things as they are in regard to reading the param...

(btw, I think we should adopt a new naming scheme: params that are initialized strictly on error, and never on access we should start prefixing with `reterr_`, to match the existing nomenclature of `ret_` for those which are initialized on success, and never on failure. I think that would make things nicely symmetric and easier to grok. thee sd_bus_error parameters should then be renamed to reterr_error or so... just an idea)

File: src/core/unit.c [hunk 2]
```diff
@@ -4108,29 +4108,29 @@ int unit_kill(
...
-                                                        "Failed to send signal SIG%s to auxiliary processes: %m",
-                                                        signal_to_string(signo));
-                                }
+                if (r < 0 && !IN_SET(r, -ESRCH, -ENOENT)) {
+                        RET_GATHER(ret, r);
 
-                                log_unit_warning_errno(
-                                                u, r,
-                                                "Failed to send signal SIG%s to auxiliary processes on client request: %m",
+                        if (!sd_bus_error_is_set(error))
```
Comment (poettering) [inline src/core/unit.c]: not a fan of this here either.

**Comment by poettering** [comment]
lgtm, just not a fan of the code that reads the bus error param, as mentioned above. we should consider it purely a return param i think which is set on failure, but not used on success.

File: src/core/unit.c (hunk 1 above)
Comment (yuwata) [reply src/core/unit.c]: So, you mean `reterr_error` here? 
In this PR, I stick to previous rule, i.e. `ret_error` here. I'd like to rename them in a later PR consistently.

File: src/core/unit.c [hunk 3]
```diff
@@ -4746,6 +4746,42 @@ static int operation_to_signal(
...
+                _cleanup_free_ char *comm = NULL;
+
+                (void) pidref_get_comm(pidref, &comm);
+                return log_unit_warning_errno(u, r, "Failed to kill %s process " PID_FMT " (%s), ignoring: %m", type, pidref->pid, strna(comm));
+        }
+
+        if (send_sighup)
+                (void) pidref_kill(pidref, SIGHUP);
+
+        return !is_alien;
```
Comment (poettering) [inline src/core/unit.c:4793]: maybe add comment: `/* return > 0 if it makes sense to wait for this process' SIGCHLD, == 0 if not */`

**Pull request merged**